### PR TITLE
Update contributing doc to suggest use of gh CLI

### PR DESCRIPTION
**Pull request by justinsb**
Also suggest running the mockgcp tests, for faster validation.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -47,39 +47,51 @@ We follow the
...
-    We recommend you to create the local clone under the path
-    `~/go/src/github.com/YOUR_USERNAME`. This will help to avoid a few known
-    build frictions related to generated code.
 
-    ```shell
-    mkdir -p ~/go/src/github.com/YOUR_USERNAME
-    cd ~/go/src/github.com/YOUR_USERNAME
-    git clone https://github.com/YOUR_USERNAME/k8s-config-connector   # If you use ssh key auth, this will be git@github.com:YOUR_USERNAME/k8s-config-connector.git
-    ```
+### Run tests against a mock environment
```
Comment (diviner524) [inline CONTRIBUTING.md:77]: Shall we move this right after the title "Validate your environment"?

https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/860/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R181

The instructions assumes Golang is already installed, which may not be the case.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -47,39 +47,51 @@ We follow the
...
+    `~/go/src/github.com/GoogleCloudPlatform`. 
 
-    https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository
+    ```shell
+    gh repo clone github.com/GoogleCloudPlatform/k8s-config-connector ~/go/src/github.com/GoogleCloudPlatform/k8s-config-connector
+    cd ~/go/src/github.com/GoogleCloudPlatform/k8s-config-connector
+    ```
 
-3.  Clone your forked repo to your dev machine.
+1. Create a fork of the repo in your github account: `gh repo fork`
```
Comment (diviner524) [inline CONTRIBUTING.md:69]: Move this to a new line?

```
gh repo fork
```

Also I think we want to mention select "Yes" for "Would you like to add a remote for the fork?" This is what I see.

```
$ gh repo fork
✓ Created fork diviner524/k8s-config-connector
? Would you like to add a remote for the fork? Yes
✓ Added remote origin
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -47,39 +47,51 @@ We follow the
...
+
+```
+# Download and set up envtest, a local test version of kubernetes apiserver
+export KUBEBUILDER_ASSETS=$(go run sigs.k8s.io/controller-runtime/tools/setup-envtest@master use -p path)
+
+RUN_E2E=1 E2E_KUBE_TARGET=envtest E2E_GCP_TARGET=mock go test ./tests/e2e/ -test.count=1 -v -run TestAllInSeries/samples/privatecacapool
+```
+
+This test takes about a minute (after compilation) - and we're working on making it much faster.  mockgcp coverage is currently pretty sparse,
+but we are working on adding coverage for more of the GCP APIs.
```
Comment (cheftako) [inline CONTRIBUTING.md:92]: Would be good to let them know we expect something like
```
=== CONT  TestAllInSeries
    unified_test.go:126: shutting down manager
--- PASS: TestAllInSeries (50.71s)
    --- PASS: TestAllInSeries/samples (50.71s)
        --- PASS: TestAllInSeries/samples/privatecacapool (48.92s)
PASS
ok  	github.com/GoogleCloudPlatform/k8s-config-connector/tests/e2e	51.313s
```

**Comment by cheftako** [comment]
Ran through the steps and other than a pre-existing issue with gcp-setup.sh, the new steps work for me.

File: CONTRIBUTING.md (hunk 1 above)
Comment (justinsb) [reply CONTRIBUTING.md:77]: I'd rather we gave positive feedback as early as possible.  We're demonstrating that it's just a go program.  I added "install golang" as a pre-requisite.

File: CONTRIBUTING.md (hunk 2 above)
Comment (justinsb) [reply CONTRIBUTING.md]: Added the `--remote` argument which does this automatically

File: CONTRIBUTING.md (hunk 3 above)
Comment (justinsb) [reply CONTRIBUTING.md:96]: Added - thank you!

**Comment by diviner524** [comment]
@justinsb 

Do we want to wait until next week to merge this? We can merge it now but it will get overwritten by the active internal -> GitHub sync.

**Comment by diviner524** [comment]
> > We can merge it now but it will get overwritten by the active internal -> GitHub sync.
> 
> This is only docs so it's easy enough to reapply if it does get overwritten.
> 
> For code though, I think we should avoid flip-flopping. Maybe we are ready to turn off the internal -> Github sync?

SG, I will merge this PR then. There are still a few pending changes from internal. We will sort out by end of week.

**Pull request merged**